gcc/rust/backend/rust-compile-pattern.cc

```diff
@@ -387,5 +387,51 @@ CompilePatternLet::visit (HIR::WildcardPattern &pattern)
     }
 }
 
+void
+CompilePatternLet::visit (HIR::TuplePattern &pattern)
+{
+  rust_assert (!ty->is_unit ());
+  rust_assert (pattern.has_tuple_pattern_items ());
+  switch (pattern.get_items ()->get_pattern_type ())
+    {
+    default:
+      gcc_unreachable ();
+    case HIR::TuplePatternItems::TuplePatternItemType::RANGED:
+      rust_fatal_error (pattern.get_locus (),
+			"ranged tuple pattern in let statement");
```

Comment:
Is this something we cannot compile yet but that should be allowed? In which case, 
```suggestion
      rust_sorry_at (pattern.get_locus (),
			"ranged tuple pattern in let statement");
```
if not, why is it a fatal_error and not a regular error? Is it a compiler error to have a ranged pattern in a tuple pattern, as if "did something go wrong earlier in gccrs to get to that point"? In that case, we should have an assertion failure IMO